base/ltoutenc.dtx

```diff
@@ -1717,6 +1717,15 @@
 \DeclareTextSymbol{\textemdash}{OT1}{124}
 \DeclareTextSymbol{\textendash}{OT1}{123}
 %    \end{macrocode}
+%    
+% \changes{v2.0u}{2021/02/19}{Add \cs{textnonbreakinghyphen}, \cs{textfiguredash}
+%      and \cs{texthorizontalbar} (gh/404)}
+%    \begin{macrocode}
+\DeclareTextCommand{\textnonbreakinghyphen}{OT1}{\mbox{-}\hskip\z@}
```

Comment:
(Affects all three definitions of `\textnonbreakinghyphen`)
Due to the `\hskip` this allows a break, making it a breaking hyphen. We should either remove the `\hskip` or suppress the break e.g. using `\penalty10000`.